In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('pacific.csv')

 ID: There are basically 1050 different ids presented here

Name: 267 different names

Date: 5231 different dates

Event: 5 different events

Status: 12 different status

Minimum Pressure: 118 

Maximum Wind: 37

Latitude:371

Longitude:1190

67 different years from 1949 to 2015

Augest,september and july are the most affected months [8,9,7]

In [3]:
df['Year'] = df['Date'].astype(str).str[:4]

In [4]:
df['Year'].value_counts().sort_values(ascending=False)[:10]

1992    890
2014    840
1990    767
1997    725
2015    717
1985    704
2006    683
1984    659
1994    654
1982    653
Name: Year, dtype: int64

In [5]:
df['Month'] = df['Date'].astype(str).str[4:6]

In [6]:
df['Month'].value_counts().sort_values(ascending=False)

08    6931
09    6206
07    5345
10    3560
06    2653
05     634
11     592
12     151
02      33
01      21
03      11
Name: Month, dtype: int64

#### since 2001 to 2015 there were 7661 disasters happened which is often 34%

In [4]:
f_2001_To_2015 = df[df['Year'].astype('int64')>2000]
f_2001_To_2015['Year'].value_counts().sort_values(ascending=False).sum()

7661

In [21]:
df.groupby('Latitude')['Maximum Wind'].count().sort_values(ascending=False)[:10]

Latitude
16.0N    337
17.0N    317
15.0N    306
15.5N    287
14.0N    287
16.7N    280
14.6N    274
17.5N    273
18.0N    270
16.3N    270
Name: Maximum Wind, dtype: int64

from 14.0N to 18.0N is highly affected 

Latitude in 16.0N there is maximum wind is 130 and minimum wind is 20 [25,45 and 30 is most] and average wind is 52.27.

and minimum pressure is 910 and maximum pressure 1014 and average is 993.29 [where after year 2000 average is 994.19] 

#### size 163 with -999

In [38]:
df[(df["Latitude"]=="16.0N") & (df["Minimum Pressure"]!=-999)]['Minimum Pressure'].sort_values(ascending=False)

13248    1014
13241    1012
23279    1011
20058    1011
20059    1011
         ... 
20955     950
22540     948
23327     946
25081     929
20878     910
Name: Minimum Pressure, Length: 163, dtype: int64

In [80]:
#16N
N16 = df[(df["Latitude"]=="16.0N")&(df['Minimum Pressure']!=-999)] 

In [91]:
N16[N16['Year'].astype('int32')>2000]['Minimum Pressure'].mean()

994.1941747572815

In [95]:
#17N
N17 = df[(df["Latitude"]=="17.0N")&(df['Minimum Pressure']!=-999)] 

### 17.0N
Minimum pressure is 933 and max pressure is 1012 and average is 993.87

In [100]:
N17['Minimum Pressure'].mean()

993.8758620689655

## All latitude

minimum pressure is 872 and maximum is  1021 and average is 994.99

maximum wind is 185 and minimum is (10 with -999 value) 15 and average wind is 48.133

In [151]:
#All
df[df['Minimum Pressure']!=-999]['Maximum Wind'].mean()

48.1332783319583

# Longitude

### top 10 affected longitude ( 108W to 112W )

In [119]:
df['Longitude'].value_counts().sort_values(ascending=False)[:10]

110.0W    135
109.0W    124
111.0W    122
112.0W    108
109.5W    105
112.5W    103
108.8W    102
108.0W    101
108.6W    101
110.8W     98
Name: Longitude, dtype: int64

### 110.0W 
minimum pressure is 935 and maximum pressure is 1010 and average is 991.46

minimum wind is 20 and maximum wind is 125 and average wind speed is 47.69

In [128]:
#110.0W
W110 = df[(df["Longitude"]=="110.0W")]['Maximum Wind'] 

In [132]:
W110.mean()

47.696296296296296

### 109.0W 
minimum pressure is 921 and maximum pressure is 1012 and average is 991.89

minimum wind is 15 and maximum wind is 140 and average wind speed is 50.56

In [138]:
W109 = df[(df["Longitude"]=="109.0W") & (df['Minimum Pressure']!=-999)]

In [141]:
W109['Minimum Pressure'].mean()

991.8909090909091

# All Longitude

minimum wind is 15 ,maximum wind is 185 and average wind is 48.133

minimum pressure 872 ,maximum pressure is 1021 and average pressure is 994.99

In [149]:
#All
df[df['Minimum Pressure']!=-999]['Minimum Pressure'].mean()

994.9949748743719

# relation between Latitude and Longitude

# Top 20 Maximum affected latitude and longitude combinations of Maximum Wind

In [155]:
df.groupby(['Latitude','Longitude'])['Maximum Wind'].value_counts().sort_values(ascending=False)[:20]

Latitude  Longitude  Maximum Wind
12.8N     157.3W     30              8
16.8N     142.0W     40              5
12.4N     145.0W     30              5
13.8N     174.0W     45              4
17.0N     110.0W     25              4
23.2N     120.4W     30              3
17.7N     114.2W     35              3
18.0N     109.0W     45              3
17.3N     109.0W     30              3
18.0N     110.0W     30              3
16.8N     110.0W     30              3
14.2N     109.6W     25              3
23.3N     143.3W     65              3
14.3N     105.7W     45              3
14.7N     157.7W     30              3
14.2N     101.0W     35              3
11.5N     97.0W      25              3
12.9N     109.8W     45              3
12.8N     95.6W      25              3
11.7N     99.6W      30              3
Name: Maximum Wind, dtype: int64

# Top 20 Maximum affected latitude and longitude combinations of Minimum Pressure

In [181]:
df[df['Minimum Pressure']!=-999].groupby(['Latitude','Longitude'])['Minimum Pressure'].value_counts().sort_values(ascending=False)[:20]

Latitude  Longitude  Minimum Pressure
19.5N     110.0W     1003                2
14.6N     108.7W     1005                2
19.0N     138.5W     1010                2
14.9N     129.9W     974                 2
14.3N     118.6W     960                 2
14.9N     104.3W     1004                2
21.0N     108.0W     1004                2
14.7N     106.9W     1006                2
14.6N     156.8W     1010                2
12.9N     109.0W     1008                2
16.2N     111.0W     1005                2
13.6N     111.5W     1000                2
20.8N     128.0W     1011                2
23.1N     115.7W     1011                2
15.4N     106.2W     1007                2
18.7N     112.2W     1006                2
12.3N     127.8W     1008                2
26.1N     118.6W     1010                2
20.0N     136.7W     1010                2
15.3N     119.3W     1009                2
Name: Minimum Pressure, dtype: int64